src/datanode/src/instance/grpc.rs

```diff
@@ -119,31 +121,47 @@ impl Instance {
         }
     }
 
-    pub async fn handle_insert(
+    pub async fn handle_inserts(
         &self,
-        request: InsertRequest,
-        ctx: QueryContextRef,
+        requests: InsertRequests,
+        ctx: &QueryContextRef,
     ) -> Result<Output> {
         let catalog = &ctx.current_catalog();
         let schema = &ctx.current_schema();
-        let table_name = &request.table_name.clone();
-        let table_ref = TableReference::full(catalog, schema, table_name);
 
-        let table = self
-            .catalog_manager
-            .table(catalog, schema, table_name)
-            .await
-            .context(CatalogSnafu)?
-            .with_context(|| TableNotFoundSnafu {
-                table_name: table_ref.to_string(),
-            })?;
-
-        let request = common_grpc_expr::insert::to_table_insert_request(catalog, schema, request)
-            .context(error::InsertDataSnafu)?;
+        let mut inserts = Vec::with_capacity(requests.inserts.len());
+        for insert in requests.inserts {
+            let table_name = &insert.table_name;
+            let table = self
+                .catalog_manager
+                .table(catalog, schema, table_name)
+                .await
+                .context(CatalogSnafu)?
+                .with_context(|| TableNotFoundSnafu {
+                    table_name: common_catalog::format_full_table_name(catalog, schema, table_name),
+                })?;
+
+            let catalog_name = catalog.clone();
+            let schema_name = schema.clone();
+            let table_name = table_name.clone();
+            inserts.push(async move {
+                let request = to_table_insert_request(&catalog_name, &schema_name, insert)
+                    .context(InsertDataSnafu)?;
+
+                table.insert(request).await.with_context(|_| InsertSnafu {
+                    table_name: common_catalog::format_full_table_name(
+                        &catalog_name,
+                        &schema_name,
+                        &table_name,
+                    ),
+                })
+            });
+        }
 
-        let affected_rows = table.insert(request).await.with_context(|_| InsertSnafu {
-            table_name: table_ref.to_string(),
-        })?;
+        let results = future::try_join_all(inserts.into_iter().map(common_runtime::spawn_write))
```

Comment:
Why not `spawn_write` above? Then the `inserts` is the vector of futures to join.